Map/MapBase.gd

```diff
@@ -0,0 +1,9 @@
+extends Resource
+class_name MapBase
+## Class for the map. Provides functions for the map
+##
+## Holds basic functionality for a map
+
+@export var rooms: Array = [RoomBase] ## 2D array of rooms on the map. this will be assigned when generating rooms
+@export var width: PackedInt32Array = [2,3,2]
```

Comment:
This is kinda "magic number" but it actually makes sense. If you look at the issue, you can see that the map has a variable width. The numbers here represent the width of successive floors. I don't think it would be good to add naming because this might be extended to more than those 3 numbers (there will be a many width as there are floors).

A comment would be good to explain that though.

Reply:
A comment for clarity on this would be fine then.